pkg/provider/consulcatalog/config.go

```diff
@@ -85,6 +102,85 @@ func (p *Provider) buildConfiguration(ctx context.Context, items []itemData) *dy
 	return provider.Merge(ctx, configurations)
 }
 
+func connectTransportName(item itemData) string {
+	return fmt.Sprintf("tls-%s-%s-%s", item.Namespace, item.Datacenter, item.Name)
+}
+
+type connectCert struct {
+	root []string
+	leaf keyPair
+	err  error
```

Comment:
let's document this, as it's bit out of place and unexpected (feel free to reword if i didn't get it right).

// err is used to propagate to the caller (Provide) any error occurring within
// the certificate watcher goroutines.